File: typescript/easy/0258.addDigits.ts
function addDigits(num: number): number {
    let sum: number = num
    let newArr: string[]

    while (sum > 9) {
        newArr = sum.toString().split("")
        sum = newArr.reduce((a, b) => Number(a) + Number(b), 0)
    }

    return sum
}

console.log(addDigits(38)) // 2

//todo Input: num = 38
//todo Output: 2
//todo Explanation: The process is
//todo 38 --> 3 + 8 --> 11
//todo 11 --> 1 + 1 --> 2
//todo Since 2 has only one digit, return it.

File: python/easy/0258.addDigits.py
class Solution:
    def addDigits(self, num: int) -> int:
        while num > 9:
            num = sum(int(digit) for digit in str(num))

        return num


if __name__ == "__main__":
    solution = Solution()
    print(solution.addDigits(38))  # 2
